src/run.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const logger = require('winston');
+const parser = require('js-yaml');
+const fs = require('fs');
+const _ = require('lodash');
+
+let spawnSync = require('child_process').spawnSync;
+
+function expandTokens(command, vars) {
+  const template = _.template(command);
+
+  return template(vars);
+}
+
+function resolveKeyValuePairs(array, vars) {
+  const templated = vars ? _.map(array, a => expandTokens(a, vars)) : array;
+  const split = _.map(templated, a => a.split('='));
+  const hashed = _.fromPairs(split);
+  return hashed;
+}
+
+function buildSpawnOptions(env) {
+  return !_.isEmpty(env) ? { env: env, stdio: 'inherit' } : { stdio: 'inherit' };
+}
+
+function execCommand(cmdTemplate, environment, vars) {
+  if(!cmdTemplate) throw new Error('Command not defined')
+  const command = expandTokens(cmdTemplate, vars).split(" ");
+  const parsedEnv = resolveKeyValuePairs(environment, vars);
+  const spawnOptions = buildSpawnOptions(parsedEnv)
+
+  logger.info(`Executing command : ${command.join(" ")}`);
+
+  return spawnSync(command[0], _.tail(command), spawnOptions)
+}
+
+module.exports = (taskName, taskVars, opts) => {
+  const usherFile = opts.filepath || '.usher.yml';
+  const parsedVars = resolveKeyValuePairs(taskVars);
+  const config = parser.safeLoad(fs.readFileSync(usherFile, 'utf8'));
+  const vars = _.merge(config.vars, parsedVars);
```

Comment:
Feels like we could merge these 4 lines into an extracted function - `buildConfig ` or something like that